### ISSUE TITLE
Potential Pending selling issue

### ISSUE THREAD

**Issue by carllewis1980** [taniman/profit-trailer#1933]
NO NEW FEATURES! Use [PT Enahancements](https://github.com/taniman/profit-trailer-enhancements) for Feature Requests.

## ProfitTrailer info
Version:

 
Exchange:
- [ ] Bittrex
- [x ] Binance
- [ ] Poloniex

 
Market:
- [ x] BTC
- [ ] ETH
- [ ] USDT
- [ ] BNB
- [ ] XMR

 
Operating System:
- [x ] Windows (please specify version)
- [ ] Ubuntu (please specify version)
- [ ] Other linux distro (please specify)
- [ ] Mac OS

## Current Behavior

Let’s say I have a bag of -3%.

I dilute my bags, by splitting it into $9 pending orders, selling at 1% above average price. Then pull them out one by one as they sell.

I have had a few instances where I have had -3% bag. I have divided it up and immediately sold every part of the bag for +3%.

I can only assume there is a mid calculation in the initial bag calculation or there is a miscalculation in the price once it is brought out of pending

Can you confirm?

## Steps to Reproduce

1.
2.
3.
4.

**Comment by taniman** [comment]
You left an important part out. What version are you using?

**Comment by taniman** [comment]
and could you provide your log where this happened?

**Comment by carllewis1980** [comment]
Currently using the latest version. But has happened on previous versions.

How do I show or email the log?

**Comment by SgtBatten** [comment]
GitHub allows you to upload files. Drag and drop it

**Comment by carllewis1980** [comment]
my log files are too big 39mb

could it be because te dust is artificially raising the value of the bag. then after it has sold the dust is still there artificially raise the price of the next bag?

**Comment by carllewis1980** [comment]
[ProfitTrailer - Copy.zip](https://github.com/taniman/profit-trailer/files/2092889/ProfitTrailer.-.Copy.zip)

**Comment by carllewis1980** [comment]
managed to zip it

**Comment by carllewis1980** [comment]
i would then be able to pull out every $7 one at a time and sell them all for 1.43%

**Comment by taniman** [comment]
Your screenshot is for Version 2.0.10

I am not sure that the calculatuon is correct in this version.
You should use the latest version  2.0.12

The bot should lose the bought price in a nornal scenario because it does not support dilution.

This calculation is too complex if you start sending coins to pending.

So correct behaviour should be:

bag -10%
send bag to pending. still -10%
pick a small piece and put in DCA. Still -10%
DCA this small piece. % Changes
Sell Small piece for profit.
Take another small piece from Pending
At this moment bot cannot calculate the correct price and should show 0.00% and ask you to set price manually.
This is the correct behaviour.

So if you use 2.0.12 and get a pending that starts selling with incorrect profit please create a new issue

**Comment by carllewis1980** [comment]
So under the current situation PT is selling my bags at a possible loss when it thinks it is at a profit?

**Comment by taniman** [comment]
If a wrong calculation is done than this possibility exists.

But most people I know that do dilution make sure to set their bought prices manually

**Comment by carllewis1980** [comment]
I tried initially to set the buy price but every time that it DCA it lost the price so it was virtually impossible - maybe this has been fixed?

The best solution would be to have a alternative to Pending by having a “Dilution” page.

I know it would be near impossible to know the purchase price of each coin or part of, it would tho be easy enough to calculate an average price.

We could set a percentage and/or and option to send it to dilution when MAX DCA is reached.

Then when a spare pair space is available it takes a initial element from the dilution bags instead of buying a new pair. The pair set would be the average price for the coin

Regards

Carl Lewis
07796 926000

Fuzzy Squares
29 East Road, Bridport, Dorset, DT6 4AG
www.FuzzySquares.com

**Comment by carllewis1980** [comment]
i have just calculated i am $300 down - i can't tell you how annoying and deflating it is.

Are you planning to have a dilute function? if so how long before it is implemented?

I can't see myself continuing to use PT until it has one

**Comment by taniman** [comment]
There are third parties planning or already have this implemented. New feature requests do not go here. But I can tell that we have started to hear more about this and have started to look into it. 

But we hve no clear plans yet

**Comment by carllewis1980** [comment]
Would you be willing to pint me in the direction of an add on that does this function or similar?
